src/auth/dto/signin-credentials.dto.ts

```diff
@@ -0,0 +1,12 @@
+import { IsString, MinLength, MaxLength, Matches } from "class-validator";
+
+export class SignInCredentialsDto {
+
+    public email: string;
+
+    public phoneNumber: string;
+
+    public username: string;
```

Comment:
we still need to unify these for example `identifier` and validate it in another way from what we are doing now

maybe try to find a user with username equal to identifier value if not try as email then number

or we can verify it by regex if it matches email regex try to find user with this email if it matches phone number try with it and same for user